index.js

```diff
@@ -129,6 +152,41 @@
       Object.prototype.toString.call(x).slice('[object '.length, -']'.length);
   }
 
-  return type;
+  //# parse :: String -> { namespace :: Nullable String, name :: String, version :: Number }
+  //.
+  //. Takes any String and parses it according to the [specification][3],
+  //. returning an object with a `namespace`, a `name` and a `version` field.
+  //.
+  //. ```javascript
+  //. > parse('my-package/List@2');
+  //. {namespace: 'my-package', name: 'List', version: 2}
+  //.
+  //. > parse('nonsense!');
+  //. {namespace: null, name: 'nonsense!', version: 0}
+  //.
+  //. > function Identity(x) { this.value = x; }
+  //. > Identity['@@type'] = 'my-package/Identity@1';
```

Comment:
We could remove these two lines as the definition from the first block of doctests should be in scope. I don't mind the repetition, but I want you to know that the repetition is not required in case you'd prefer to avoid it.